wordpress-mobile/WordPress-Android
# Jetpack Mobile: Threat Details - list items builder

Parent: #13326

This PR builds list items for threat details screen.
Custom views for context lines, file name are created in a child PR: #13704


### To test 

Prerequisite: 
1. Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
2. WP.com account with a site having scan capability and different threats (e.g. http://do.wpmt.co/jp-scan-daily/).

- Login to the app using above WP.com account and select the site on the My Site tab
- Notice that scan menu is displayed
- Click scan menu item followed by threat list item
- Notice that threat details list items are built correctly as per the table below.
(Tests included in `ThreatDetailsListItemsBuilderTest`)

References: 
1. Calypso's [threat description](https://opengrok.a8c.com/source/xref/calypso/client/components/jetpack/threat-description/index.tsx?r=e940ccc1#79), [fix details](https://opengrok.a8c.com/source/xref/calypso/client/components/jetpack/threat-item/index.tsx?r=d1e731dd#83) and  [buttons](https://opengrok.a8c.com/source/xref/calypso/client/components/jetpack/threat-item/index.tsx?r=d1e731dd#162).
2. [Scan design iteration i5 ](https://jetpackmobile.files.wordpress.com/2020/12/ios-scan-i5.png)

Basic Details
Item | Icon/Text |
-- | --
Icon | ic_scan_idle_threats_found
ThreatHeader | Same as ThreatItem Header (in red) #13646
ThreatSubHeader | Same as ThreatItem SubHeader #13646
ProblemHeader | What was the problem?
ProblemDescription | _{threatModel.baseThreatModel.description}_

Technical Details

Item | Title
-- | --
Header | The technical details |

Item |  Present in
-- | ---
Custom View for fileName | CoreFileModificationThreatModel, FileThreatModel
Custom View for threatContext | FileThreatModel
 diff | CoreFileModificationThreatModel


Fix Details

Item | ThreatStatus | Fixable? | Text
-- | -- | -- | --
Header | FIXED |   | How did Jetpack fix it?
  | CURRENT | Yes | How will we fix it?
  |   | No | Resolving the threat
  | IGNORED, UNKNOWN |   | How will we fix it?

Item | Fixable? |  FixType | Text
-- | -- | -- | --
Description | Yes | REPLACE | Jetpack Scan will replace the affected file or directory.
  |   | DELETE | Jetpack Scan will delete the affected file or directory.
  |   | UPDATE | If fixable.target presentJetpack Scan will update to a newer version {fixable.target}.ElseJetpack Scan will resolve the threat.
  |   | EDIT | Jetpack Scan will edit the affected file or directory.
  |   | UNKNOWN | Jetpack Scan will resolve the threat.
  | No |   | Jetpack Scan cannot automatically fix this threat. We suggest that you resolve the threat manually: ensure that WordPress, your theme, and all of your plugins are up to date, and remove the offending code, theme, or plugin from your site. \n \n If you need more help to resolve this threat, we recommend <b>Codeable</b>, a trusted freelancer marketplace of highly vetted WordPress experts. They have identified a select group of security experts to help with these projects. Pricing ranges from $70&#8211;120/hour, and you can get a free estimate with no obligation to hire.

Buttons

| ThreatStatus | Fixable? | Button
-- | -- | --
ThreatStatus != FIXED | Yes | Fix threat
  | No | Get a free estimate
CURRENT |   | Ignore threat

### Notes
1. Ignore all styling


### Merge Instructions

1. ~~Make sure #13675, child PR #13704 are merged~~ Merged
2. ~~Update FluxC hash with tag for wordpress-mobile/WordPress-FluxC-Android#1819~~ Done in 21e0436
3. ~~Remove the "Not Ready for Merge label"~~ Removed
4. Merge this PR

PR submission checklist:
